sidecar/supervisor/appsupervisor.go

```diff
@@ -170,16 +186,43 @@ func terminateSubprocesses(procs []*process, sig os.Signal) {
 			}
 		}
 
+		// If services have not exited gracefully within the alotted time, force kill them all
 		if time.Now().After(timeout) {
 			for _, proc := range procs {
 				if proc.Cmd.Process != nil {
-					proc.Cmd.Process.Kill()
+					syscall.Kill(-proc.Cmd.Process.Pid, syscall.SIGKILL)
 				}
 			}
+			// Allow for processes to handle signal
+			time.Sleep(100 * time.Millisecond)
```

Comment:
Child processes *cannot* handle SIGKILL, as it cannot be trapped. For my understanding, the OS kills them immediately, so this sleep isn't necessary.